### ISSUE TITLE
java version mismatch breaks solr in ci

### ISSUE THREAD

**Issue by thatbudakguy** [Princeton-CDH/ppa-django#313]
from travis logs:
```
Starting solr and creating core test-ppa...
Your current version of Java is too old to run this version of Solr
We found version 11.0.2, using command '/usr/local/lib/jvm/openjdk11/bin/java -version', with response:
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```
unlikely that a build from january of this year is "too old", so there's probably something wrong with the logic solr uses to check the current java version.

**Comment by thatbudakguy** [comment]
## postmortem
travis builds are now on ubuntu xenial, which installs and uses `openjdk11` by default. solr 6.x is either unaware of jdk versions as high as 11, or has a [bug in the logic](https://stackoverflow.com/a/49600216) it uses to determine what is a new enough jdk version. some [posts by maintainers](http://lucene.472066.n3.nabble.com/Solr-6-6-and-OpenJDK11-td4430683.html) also indicate that versions of the jdk >=9 are only recommended for solr 7.x and above.
## solution
if a project does not have `language:java` set, specifying e.g. `jdk:openjdk8` _installs_ but does not select `openjdk8`. travis's trusty image included a `jdk-switcher` script, but it's not included in xenial. therefore, we have to specify that `openjdk8` should be installed and set `JAVA_HOME` to point to`openjdk8`.
```yaml
jdk:
  - openjdk8
env:
  global:
    - JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-amd64
```
## notes
this issue could definitely recur: we are on the oldest supported version of solr, and java is now on a 6-month release schedule, meaning the jdk versions will only get more and more out of step with us. this forum poster sums it up well:
> There is a lack of consensus about Java 11 support. We have been recommended to stick to Java 8 even on Solr 7.X. Is the page bellow the 'official' position? 

the [page they mention](https://cwiki.apache.org/confluence/display/solr/SolrJavaVersions) that supposedly recommended older solr versions stick to jdk8 is no longer maintained; and current versions of the solr manual don't address this except for versions of solr > 7. it seems like the safest option is to move towards newer solr versions as soon as we can.

**Comment by thatbudakguy** [comment]
resolved by #338

**Comment by rlskoeser** [comment]
was marked as resolved but not closed! (fixed by switching from travis-ci to github actions for CI)
